pkg/nodeconfig/payload/payload.go

```diff
@@ -161,9 +167,86 @@ if( $endpoint -eq $null) {
     $endpoint = New-HnsEndpoint -NetworkId $hns_network.ID -Name "VIPEndpoint"
     Attach-HNSHostEndpoint -EndpointID $endpoint.ID -CompartmentID 1
 }
+# Get HNS endpoint IP
+$ipv4Address = (Get-NetIPConfiguration -AllCompartments -All -Detailed | Where-Object { $_.NetAdapter.LinkLayerAddress -eq $endpoint.MacAddress }).IPV4Address.IPAddress.Trim()
```

Comment:
does the script still return this value? if not, please update the comment at the top of the script with the new output

Reply:
fixed!